tests/test_init.py

```diff
@@ -0,0 +1,162 @@
+        'tap_dummy_test-test_table_one': {
+            'target_key': 'tap_dummy_test-test_table_one-20210811T063938.jsonl',
+            'file_name': Path('tests/output/tap_dummy_test-test_table_one-20210811T063938.jsonl'),
+            'file_data': []},
+        'tap_dummy_test-test_table_two': {
+            'target_key': 'tap_dummy_test-test_table_two-20210811T063938.jsonl',
+            'file_name': Path('tests/output/tap_dummy_test-test_table_two-20210811T063938.jsonl'),
+            'file_data': []},
+        'tap_dummy_test-test_table_three': {
+            'target_key': 'tap_dummy_test-test_table_three-20210811T063938.jsonl',
+            'file_name': Path('tests/output/tap_dummy_test-test_table_three-20210811T063938.jsonl'),
+            'file_data': [
+                '{"c_pk": 1, "c_varchar": "1", "c_int": 1, "c_time": "04:00:00"}\n',
+                '{"c_pk": 2, "c_varchar": "2", "c_int": 2, "c_time": "07:15:00"}\n',
+                '{"c_pk": 3, "c_varchar": "3", "c_int": 3, "c_time": "23:00:03", "_sdc_deleted_at": "2019-02-10T15:51:50.215998Z"}\n']}}
+
+
+def test_emit_state(capsys, state):
+    '''TEST : simple emit_state call'''
+
+    emit_state(state)
+    captured = capsys.readouterr()
+    assert captured.out == json.dumps(state) + '\n'
+
+
+def test_float_to_decimal():
+    '''TEST : simple float_to_decimal call'''
+
+    assert float_to_decimal({
+        "type": "RECORD", "stream": "tap_dummy_test-test_table_one",
+        "record": {
+            "c_pk": 1, "c_varchar": "1", "c_int": 1, "c_float": 1.99},
+        "version": 1, "time_extracted": "2019-01-31T15:51:47.465408Z"}) == {
+            "type": "RECORD", "stream": "tap_dummy_test-test_table_one",
+            "record": {
+                "c_pk": 1, "c_varchar": "1", "c_int": 1, "c_float": Decimal('1.99')},
+            "version": 1, "time_extracted": "2019-01-31T15:51:47.465408Z"}
+
+
+def test_get_target_key():
+    '''TEST : simple get_target_key call'''
+
+    assert get_target_key({'stream': 'dummy_stream'}, naming_convention='{stream}-{timestamp}.jsonl', timestamp='99') == 'dummy_stream-99.jsonl'
+
+
+def test_save_file(config, file_metadata):
+    '''TEST : simple save_file call'''
+    Path(config['temp_dir']).mkdir(parents=True, exist_ok=True)
+    for _, file_info in file_metadata.items():
+        save_file(file_info, 'none')
+
+    assert not file_metadata['tap_dummy_test-test_table_one']['file_name'].exists()
+    assert not file_metadata['tap_dummy_test-test_table_two']['file_name'].exists()
+    assert file_metadata['tap_dummy_test-test_table_three']['file_name'].exists()
+
+    clear_dir(Path(config['temp_dir']))
+
+
+def test_upload_files(monkeypatch, config, file_metadata):
+    '''TEST : simple upload_files call'''
+
+    monkeypatch.setattr(s3, 'create_client', lambda config: None)
+
+    monkeypatch.setattr(s3, 'upload_file', lambda filename, s3_client, bucket, s3_key,
+                    encryption_type=None, encryption_key=None: None)
```

Comment:
Instead of monkey patching or mocking boto3/AWS, you can use the `moto` library to setup all the mocks for you. http://docs.getmoto.org/en/latest/index.html